pkg/client/connector/traffic_manager.go

```diff
@@ -150,11 +151,17 @@ func (tm *trafficManager) run(c context.Context) error {
 		}
 	}()
 
-	conn, err = grpc.DialContext(tc, grpcAddr,
-		grpc.WithContextDialer(grpcDialer),
+	opts := []grpc.DialOption{grpc.WithContextDialer(grpcDialer),
 		grpc.WithInsecure(),
 		grpc.WithNoProxy(),
-		grpc.WithBlock())
+		grpc.WithBlock()}
+
+	if mxRecvSize := clientConfig.Grpc.MaxReceiveSize; mxRecvSize != nil {
+		if mz, ok := mxRecvSize.AsInt64(); ok {
```

Comment:
I think it would maybe make sense to do this check when we load the config (and have MaxReceiveSize be an Int64 instead of a quantity). Just so we don't need to do this check every time we use this value.. and then can also throw an error when loading the config and inform the user then.. instead of having it load successfully into the config but then not be used (Bc it's a decimal or something) which might be confusing to the user.

Reply:
Yep. That makes a lot of sense.